### PR TITLE
Refactor the editorconfig file

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -10,11 +16,19 @@ root = true
...
+[*.md]
+trim_trailing_whitespace = false
+
 # Code files
 [*.{cs,csx,vb,vbx}]
+charset = utf-8-bom
 indent_size = 4
 insert_final_newline = true
-charset = utf-8-bom
+max_line_length = 150
```
Comment (valhristov) [inline .editorconfig]: I thought we are standardizing on ~130

File: .editorconfig [hunk 2]
```diff
@@ -96,3 +124,77 @@ csharp_new_line_before_catch = true
...
+csharp_indent_switch_labels = true
+csharp_indent_labels = one_less_than_current
+# undocumented
+csharp_indent_block_contents = true
+csharp_indent_braces = false
+csharp_indent_case_contents_when_block = true
+
+# Spaces
+csharp_space_after_cast = false
+csharp_space_after_keywords_in_control_flow_statements = true
```
Comment (valhristov) [inline .editorconfig]: why `true`?

File: .editorconfig [hunk 3]
```diff
@@ -96,3 +124,77 @@ csharp_new_line_before_catch = true
...
+csharp_space_after_colon_in_inheritance_clause = true
+csharp_space_before_colon_in_inheritance_clause = true
+csharp_space_after_comma = true
+csharp_space_before_comma = false
+csharp_space_after_dot = false
+csharp_space_before_dot = false
+csharp_space_after_semicolon_in_for_statement = true
+csharp_space_before_semicolon_in_for_statement = false
+# Extra space before equals sign DOES MATTER https://github.com/dotnet/roslyn/issues/20355
+csharp_space_around_binary_operators  = before_and_after
```
Comment (valhristov) [inline .editorconfig]: 🥇

File: .editorconfig [hunk 4]
```diff
@@ -96,3 +124,77 @@ csharp_new_line_before_catch = true
...
+csharp_space_around_declaration_statements = ignore
```
Comment (Evangelink) [inline .editorconfig]: Maybe I should put

**Pull request merged**